Added exists-maven-plugin.

MMaiero opened the pull request:
Plugins added both in manifest_pom and target-platform poms.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

cdealti commented on kura/manifest_pom.xml:
```diff
@@ -119,6 +99,8 @@
         <bnd.baseline.version>3.4.0-SNAPSHOT</bnd.baseline.version>
         <bnd.baseline.continueOnError>false</bnd.baseline.continueOnError>
+
+        <maven.deploy.skip.prop>false</maven.deploy.skip.prop>
```
``maven.deploy.skip.prop`` -> ``exists.artifact.already.released``

cdealti commented on kura/manifest_pom.xml:
```diff
@@ -510,44 +491,76 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
needs build and test dependency CQ (I will open an issue)

cdealti commented on kura/manifest_pom.xml:
```diff
@@ -510,44 +491,76 @@
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
CQ build and test dependency needed

cdealti commented on kura/manifest_pom.xml:
```diff
@@ -510,44 +491,76 @@
+                <version>1.4.1</version>
+                <executions>
+                    <execution>
+                        <id>enforce-property</id>
```
``enforce-property`` -> ``fail-on-artifact-already-released``

cdealti commented on kura/manifest_pom.xml:
```diff
@@ -510,44 +491,76 @@
+                            <rules>
+                                <requireProperty>
+                                    <property>maven.deploy.skip.prop</property>
+                                    <message>Property must be false during execution.</message>
```
Artifact already exists on the release repository. Forgot to bump version?

cdealti commented on target-platform/pom.xml:
```diff
@@ -2,91 +2,103 @@
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <maven.deploy.skip.prop>false</maven.deploy.skip.prop>
```
same as above

MMaiero committed:
Added exists-maven-plugin version 0.0.3

Plugin added both in manifest_pom and target-platform poms.
The build will fail if an artifact already exists remotely.

Signe-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero commented:
Absolutely, the plugins need to be added to the build and test dependency doc before merging the PR. At least we don't miss it.

cdealti commented:
Please rebase & update the PR as the build has been fixed by #1441

cdealti commented:
@MMaiero failing the build should be avalable out of box with version 0.0.3:

https://github.com/chonton/exists-maven-plugin/issues/1

MMaiero commented:
Plugin updated to Version 0.0.3.
The build fails if an artifact already exists remotely

MMaiero commented:
Rebased branch.
Created issue for the Build and test dependency.
@pierantoniomerlino I think that if it is ok for you, we can merge.
